setup.py

```diff
@@ -107,7 +107,17 @@ def run(self):
         import build_sphinx
         build_sphinx.rebuild()
 
-if 'clean' not in sys.argv:
+# determine if this run requires building of Qt GUI ui->py
+build_qt = True
+# _standard_ commands which shouldn't trigger the Qt build
+no_build_commands = ['clean', 'sdist', 'bdist', 'check', 'upload', 'register', '--help']
```

Comment:
Use a whitelist rather than a blacklist.  First, there are a lot of commands available to setup that you have not covered (e.g., --author). Second, you can issue multiple commands in the same line, such as `python setup.py clean build`.